graphqlapi/src/main/java/io/stargate/graphql/web/resources/schemafirst/FilesResource.java

```diff
@@ -0,0 +1,164 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.graphql.web.resources.schemafirst;
+
+import com.datastax.oss.driver.shaded.guava.common.base.Charsets;
+import com.google.common.io.Resources;
+import graphql.schema.idl.SchemaPrinter;
+import io.stargate.auth.AuthenticationService;
+import io.stargate.auth.AuthenticationSubject;
+import io.stargate.auth.AuthorizationService;
+import io.stargate.auth.SourceAPI;
+import io.stargate.auth.UnauthorizedException;
+import io.stargate.auth.entity.ResourceKind;
+import io.stargate.graphql.persistence.schemafirst.SchemaSource;
+import io.stargate.graphql.persistence.schemafirst.SchemaSourceDao;
+import io.stargate.graphql.schema.scalars.CqlScalar;
+import io.stargate.graphql.web.RequestToHeadersMapper;
+import io.stargate.graphql.web.resources.Authenticated;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.GET;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+@Singleton
+@Path(ResourcePaths.FILES)
+@Authenticated
+public class FilesResource {
+
+  private final SchemaSourceDao schemaSourceDao;
+  private final AuthenticationService authenticationService;
+  private final AuthorizationService authorizationService;
+
+  @Inject
+  public FilesResource(
+      SchemaSourceDao schemaSourceDao,
+      AuthenticationService authenticationService,
+      AuthorizationService authorizationService) {
+    this.schemaSourceDao = schemaSourceDao;
+    this.authenticationService = authenticationService;
+    this.authorizationService = authorizationService;
+  }
+
+  private static final String DIRECTIVES_RESPONSE = buildDirectivesResponse();
+
+  @GET
+  @Path("/cql_directives.graphql")
+  @Produces(MediaType.TEXT_PLAIN)
+  public Response getCqlDirectives() {
+    return Response.ok(DIRECTIVES_RESPONSE)
+        .header("Content-Disposition", "inline; filename=\"cql_directives.graphql\"")
+        .build();
+  }
+
+  @GET
+  @Path("/namespace/{namespaceName}.graphql")
+  @Produces(MediaType.TEXT_PLAIN)
+  public Response getSchema(
+      @HeaderParam("X-Cassandra-Token") String token,
+      @PathParam("namespaceName") String namespace,
```

Comment:
so in the Namespace resource you test this against a pattern in order to prevent XSS attack, but here you are not doing this.. Anything different here?

Reply:
Nope, that's an oversight. I'll add the check here too.